### PR TITLE
made some changes

### PR DESCRIPTION
**Pull request by luxkatana**
I've made the follow changes and features:

- using the rich package for printing out (because it more prettier)
- checks if the data in config.json are ok
- validating the webhook given in config.json
- support for webhook_ping as <@714149216787628075> as well as you can set "714149216787628075" for the webhook_ping
- did some formatting
- did a very littlebit type hinting

***PS: I contribute to random projects 👍***

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,30 +1,78 @@
-import json, os, time, requests, cloudscraper
-from win10toast import ToastNotifier
+import cloudscraper
```
Comment (luxkatana) [inline main.py:1]: reformatting the imports

File: main.py [hunk 2]
```diff
@@ -1,30 +1,78 @@
-import json, os, time, requests, cloudscraper
-from win10toast import ToastNotifier
+import cloudscraper
+import json
+import os
+import requests
+import time
 from discord_webhook import DiscordWebhook, DiscordEmbed
-
+from rich.console import Console
```
Comment (luxkatana) [inline main.py:7]: using the **rich** package for adding colors for printing out

File: main.py [hunk 3]
```diff
@@ -1,30 +1,78 @@
...
 from discord_webhook import DiscordWebhook, DiscordEmbed
-
+from rich.console import Console
+from win10toast import ToastNotifier
 os.system(f'title Bloxflip Rain Notifier ^')
-
+console = Console()
 with open("config.json", "r") as config:
   config = json.load(config)
+  config_keys: set[str] = {key for key in config}
```
Comment (luxkatana) [inline main.py:13]: Validate if the keys exist in ``config.json`` if not it will list to the output which keys/options are missing

**Comment by aaamraj** [comment]
Hello luxkatana i'm receiving this error after running main.py
image

**Comment by luxkatana** [comment]
> Hello luxkatana i'm receiving this error after running main.py image

Hello aamraj, thanks for responding back! I don't have any idea what the problem is, I only changed the fancy things as text color and checks.

Perhaps that there is something missing in the ``config.json``? 

If that is not the case, dm me on discord, we can check together what the issue is.

Kisses,
luxkatana

**Comment by aaamraj** [comment]
> luxkatana

what is your discord username?

**Comment by luxkatana** [comment]
> > luxkatana
> 
> what is your discord username?
> 

thedatahorse

**Pull request closed without merging**